nodeup/pkg/model/containerd.go

```diff
@@ -419,3 +426,34 @@ func (b *ContainerdBuilder) buildCNIConfigTemplateFile(c *fi.ModelBuilderContext
 		Type:     nodetasks.FileType_File,
 	})
 }
+
+func (b *ContainerdBuilder) buildContainerdConfig() string {
+	cluster := b.Cluster
+
+	if cluster.Spec.ContainerRuntime != "containerd" {
+		return ""
+	}
+
+	containerd := cluster.Spec.Containerd
```

Comment:
It would be better to pass necessary values, such as `RegistryMirrors` in the `NodeupConfig` than refer to the cluster spec. I'm trying to remove references to the cluster spec from nodeup as changes to cluster spec don't mark nodes as NeedUpdate.

Reply:
I just passed in the entire struct. All of them should trigger NeedUpdate.